dsc_lib/src/functions/min.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::DscError;
+use crate::configure::context::Context;
+use crate::functions::{AcceptedArgKind, Function};
+use serde_json::Value;
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct Min {}
+
+impl Function for Min {
+    fn min_args(&self) -> usize {
+        1
+    }
+
+    fn max_args(&self) -> usize {
+        usize::MAX
+    }
+
+    fn accepted_arg_types(&self) -> Vec<AcceptedArgKind> {
+        vec![AcceptedArgKind::Number, AcceptedArgKind::Array]
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        debug!("min function");
+        if args.len() == 1 {
+            if let Some(array) = args[0].as_array() {
+                find_min(array)
+            }
+            else {
+                Err(DscError::Parser("Array cannot be empty".to_string()))
+            }
+        }
+        else {
+            find_min(args)
+        }
+    }
+}
+
+fn find_min(args: &[Value]) -> Result<Value, DscError> {
+    let array = args.iter().map(|v| v.as_i64().ok_or(DscError::Parser("Input must only contain integers".to_string()))).collect::<Result<Vec<i64>, DscError>>()?;
+    let value = array.iter().min().ok_or(DscError::Parser("Unable to find min value".to_string()))?;
+    Ok(Value::Number((*value).into()))
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::configure::context::Context;
+    use crate::parser::Statement;
+
+    #[test]
+    fn list() {
+        let mut parser = Statement::new().unwrap();
+        let result = parser.parse_and_execute("[min(3,2,5,4)]", &Context::new()).unwrap();
```

Comment:
I believe the integers are signed, so you should have one negative number in one of the tests to verify that it works correctly